### PR TITLE
add safeguard to compile only when sufficient CPU available

### PR DESCRIPTION
**Pull request by reggaemuffin**

### CODE REVIEW

**Comment by daboross** [comment]
Looks good to me! I left things like this out initially thinking that it'd help promote [making custom initialization headers per project](https://github.com/daboross/screeps-in-rust-via-wasm/blob/master/cargo-screeps/docs/initialization-header.md), but having this one check in can't hurt.

File: cargo-screeps/resources/default_initialization_header.js [hunk 1]
```diff
@@ -2,6 +2,12 @@
 let wasm_module = null;
 
 function wasm_initialize() {
+    
+    if(Game.cpu.bucket < 500){
```
Comment (daboross) [inline cargo-screeps/resources/default_initialization_header.js]: We could have some spacing around the parens here, but that can always be changed afterwards.

**Comment by reggaemuffin** [comment]
Yeah I already checked the cargo-web prepend-js directives but they append outside that wasm-init function, so they are not that useful. TBH the message was kinda uncreative 😆

File: cargo-screeps/resources/default_initialization_header.js (hunk 1 above)
Comment (reggaemuffin) [reply cargo-screeps/resources/default_initialization_header.js]: you can also edit the pr directly from the web interface if you want to fix style issues :)

**Comment by daboross** [comment]
Fair - it's built that way! The `wasm_initialize` is meant to have the most control, so that someone overriding `initialization-header` can change as many things as they like like this per-project.

I'll wait a bit to see if @ASalvail has any objections, but then we can just merge it.

**Comment by ASalvail** [comment]
Since this is somewhat needed due to the use of rust, I say it has its place in this repo.

Thanks @reggaemuffin !

Also, it might be nice to start another repo of tools and goodies we can build on top of this interface to make life easier. Not AI per se, but quality of life that goes beyond simply interfacing. Food for thought.

**Pull request merged**